.github/workflows/ci.yml

```diff
@@ -108,6 +108,54 @@ jobs:
           QSERV_DH_USER: ${{ secrets.FM_DOCKERHUB_TOKEN_USER }}
           QSERV_DH_TOKEN: ${{ secrets.FM_DOCKERHUB_TOKEN }}
 
+  documentation:
+    name: Documentation
+    runs-on: ubuntu-20.04
+    needs: update-base-images
+    steps:
+
+      - name: Install python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+
+      - name: Install click
+        run: |
+          python -m pip install --upgrade pip
+          pip install click pyyaml requests
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # 0 is "all history and branch tags"
+          submodules: true
+
+      - name: Check for changes
+        uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+              docs:
+                - '**'
+
+      - name: Prepare user build image
+        if: needs.update-base-images.outputs.build-rebuilt == 'True' || steps.filter.outputs.docs == 'true'
+        run: |
+          ./admin/local/cli/qserv --log-level DEBUG build-user-build-image \
+          --group docker_outer
+
+      - name: Build and publish
+        if: needs.update-base-images.outputs.build-rebuilt == 'True' || steps.filter.outputs.docs == 'true'
```

Comment:
only build docs if the build image was changed? That seems like a strange trigger.

Reply:
There's an "or" ("||") on that condition -- rebuild docs if base image (e.g. sphinx tools) changed, OR any files in the the docs filter have changed.